src/main/java/net/openhft/chronicle/wire/VanillaFieldInfo.java

```diff
@@ -134,7 +117,12 @@ public double getDouble(Object value) {
     public void set(Object object, Object value) throws IllegalArgumentException {
         Object value2 = ObjectUtils.convertTo(type, value);
         try {
-            getField().set(object, value2);
+            if (type.isPrimitive()) {
+                // use reflection as it does unboxing
+                getField().set(object, value);
```

Comment:
This prevents segfaults if someone erroneously calls set(...) for a primitive field